account-kit/smart-contracts/src/ma-v2/client/client.test.ts

```diff
@@ -1345,6 +1348,128 @@ describe("MA v2 Tests", async () => {
     client.setAutomine(true);
   });
 
+  it("tests entity id and nonce selection", async () => {
+    let newClient = (await givenConnectedProvider({ signer, salt: 1n }))
+      .extend(deferralActions)
+      .extend(installValidationActions);
+
+    await setBalance(client, {
+      address: newClient.getAddress(),
+      value: parseEther("2"),
+    });
+
+    const entryPoint = newClient.account.getEntryPoint();
+    const entryPointContract = getContract({
+      address: entryPoint.address,
+      abi: entryPoint.abi,
+      client,
+    });
+
+    // entity id and nonce selection for undeployed account
+    for (let startEntityId = 0; startEntityId < 5; startEntityId++) {
+      for (let startNonce = 0n; startNonce < 5n; startNonce++) {
+        const { entityId, nonce } = await newClient.getEntityIdAndNonce({
+          entityId: startEntityId,
+          nonce: startNonce,
+          isGlobalValidation: true,
+        });
+
+        const expectedEntityId: number = Math.max(1, startEntityId);
+
+        // account not deployed, expect to get 1 when we pass in 0
+        assert(entityId === expectedEntityId);
+        assert(
+          nonce ===
+            (await entryPointContract.read.getNonce([
+              newClient.account.address,
+              buildFullNonce({
+                nonce: startNonce,
+                entityId: expectedEntityId,
+                isDeferredAction: true,
+              }),
+            ]))
```

Comment:
You could also use `expect`/`toEqual` here, but for checking async results you can get even better error messages with 
```ts
await expect(entryPointContract.read.getNonce(...))
  .resolves
  .toEqual(newClient.account.address);
```
(Don't forget the `await` in front)